.github/workflows/nightly-ci-multiprocess-gpu.yml

```diff
@@ -1,77 +1,99 @@
-  pull_request: # Automatically trigger on pull requests affecting this file
-    branches:
-      - main
-    paths:
-      - '**workflows/nightly-ci-multiprocess-gpu.yml'
 jobs:
-  jaxlib-nightly:
-    runs-on: self-hosted
+  tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+    env:
+      # Secrets to connect and authenticate with the Cluster manager web API
+      WEB_API_TOKEN: ${{ secrets.WEB_API_TOKEN }}
+      WEB_API_URL:   ${{ secrets.WEB_API_URL }}
+      
+      GITHUB_WORKSPACE_REMOTE: "/home/opc/jax_ci_${{ github.run_id }}_${{ github.run_attempt }}/"
+      CONFIG: "-F ${{ github.workspace }}/.ssh/config -o UserKnownHostsFile=${{ github.workspace }}/.ssh/known_hosts"
     steps:
-      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # ratchet:actions/checkout@v3
-      - name: Launch slurm job and hook output to this shell
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
+
+      - name: Setup credentials, spin cluster up, setup test environment
+        id: setup
         run: |
-          export JOBSCRIPTSDIR=${GITHUB_WORKSPACE}/.github/workflows/slurm_job_scripts
-          source $JOBSCRIPTSDIR/slurm_utils_common.sh
-          sbatch -N 2 $JOBSCRIPTSDIR/multinode_pytest_jaxlib_nightly.sub | tee output.log
-          sleep 2m
-          export SLURM_JOBID=$(grep 'Submitted batch job' "output.log" | awk '{ print $4 }')
-          export SLURM_OUTPUT=$(scontrol show job "${SLURM_JOBID}" | grep 'StdOut' | awk -F '=' '{ print $2 }')
-          job_wait "${SLURM_JOBID}" & PID=$!
-          touch "${SLURM_OUTPUT}"
-          echo -e " ---------------------------------------------------\n" \
-               "----------WAITING FOR SLURM JOB TO BEGIN-----------\n" \
-               "---------------------------------------------------\n"
-          tail --pid="${PID}" -f "${SLURM_OUTPUT}"
-          export SLURM_STATE=$(job_state "${SLURM_JOBID}"); echo "SLURM_JOBID=${SLURM_JOBID} SLURM_STATE='${SLURM_STATE}'"
-          export SLURM_WALLTIME=$(job_time "${SLURM_JOBID}"); echo "SLURM_WALLTIME=${SLURM_WALLTIME} secs"
-          export SLURM_EXITCODE=$(job_exit_code "${SLURM_JOBID}" || echo $?); echo "SLURM_EXITCODE='${SLURM_EXITCODE}'"
-          if [ "${SLURM_EXITCODE}" != "0" ];      then exit ${SLURM_EXITCODE:-999}; fi
-          if [ "${SLURM_STATE}" != "COMPLETED" ]; then exit 1; fi
-      - name: Publish Test Results
-        uses: EnricoMi/publish-unit-test-result-action@46ab8d49369d898e381a607119161771bc65c2a6 # ratchet:EnricoMi/publish-unit-test-result-action@v2
+
+          # Create SSH keys
+          mkdir -p $GITHUB_WORKSPACE/.ssh && chmod 700 $GITHUB_WORKSPACE/.ssh
+          ssh-keygen -N '' -f $GITHUB_WORKSPACE/.ssh/id_rsa
+          PUBKEY=$(cat $GITHUB_WORKSPACE/.ssh/id_rsa.pub)
+
+          # Setup cluster, get IP address of head node, create SSH config
```

Comment:
I got rid of `GITHUB_WORKSPACE` throughout, I agree it's more readable as it it. I didn't `cd ${GITHUB_WORKSPACE}/.github/workflows/slurm_job_scripts/` since I would need to put that everywhere; I think it's pretty readable as it is, let me know.